src/parsing/compiledExpressionCache.ts

```diff
@@ -186,6 +211,9 @@ export function storeStaticCompilationResultInCache(
 	cachesForLanguage.push(
 		new CacheEntry(
 			executionStaticContext.getReferredNamespaces(),
+			typeof namespaceResolver === 'function'
+				? null
```

Comment:
Should we not store the namespace resolver in the cache either way? It seems to me the `areNamespaceResolversEqual` will also handle functions which I don't will work if we don't save the function.

Reply:
The function will not be used later on: we only reference compare the object variant.

For the function variant, we always use referred namespaces and test compatibility with the _new_ namespace resolver function.